### ISSUE TITLE
[MIOpen] Issue building CK using latest commit

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/composable_kernel#621]
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/1e59eb3be5a970ad3996d52984919a9edd7ce58f/include/ck/utility/amd_wmma.hpp#L9

If MIOpen change the line in `requirements.txt`:
```
ROCmSoftwarePlatform/composable_kernel@2045c53325e9a69f19211d105bf444efa8405b9d -DGPU_TARGETS="gfx900;gfx906;gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102"
```
and try to build CK as a dependency via:
```
mkdir -p install_dir2 && cmake -P install_deps.cmake --minimum --prefix /root/MIOpen/install_dir2
```

Then I am getting error related to the line listed on top of this issue:
```
In file included from /composable_kernel/test/wmma_op/wmma_op.cpp:14:
In file included from /composable_kernel/test/wmma_op/wmma_op_util.hpp:13:
/composable_kernel/include/ck/utility/amd_wmma.hpp:45:13: error: '__builtin_amdgcn_wmma_f32_16x16x16_bf16_w32' needs target feature gfx11-insts
            __builtin_amdgcn_wmma_f32_16x16x16_bf16_w32(
            ^
```
**[Question]**:

- When `wmma` is built for gfx11 platforms **only**, why would I meet the above issue?
- After discussion with @illsilin  it seems that the following variable is **not honored** by the compiler?
   `-DGPU_TARGETS="gfx900;gfx906;gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102"`

**Comment by aska-0096** [comment]
It seems the either arch limitation here
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/1e59eb3be5a970ad3996d52984919a9edd7ce58f/test/CMakeLists.txt#L59-L61
or compiler flag `-DGPU_TARGETS` not worked.
I agree with illia's opinion that the latter one is the cause, as gfx1100 architecture test `test_wmma_op` disabled as expected in CK CI.

**Comment by illsilin** [comment]
It looks like we have a workaround available:
replace GPU_TARGETS="gfx1030;gfx1100;gfx1101"
with
-DCMAKE_CXX_FLAGS=" --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1101"
Got a successful CI build with this workaround:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2Fcomposable_kernel/detail/lwpck-643/4/pipeline

**Comment by illsilin** [comment]
I think what's happening is that the only protection we have for the code that uses gfx11 builtins in Include/ck/utility/amd_wmma.hpp is that condition in the /test/CMakeLists.txt, which basically means "only build the wmma_op folder if gfx1100 is among the targets". But if there are multiple targets and gfx1100 is among them, the code still gets built and that's what causes the problem, because those builtins are only available on gfx11 devices. So what we need is to insert the "#if defined(__gfx11__)" conditions throughout the code to make sure those builtins are never getting invoked on any other architecture.

**Comment by aska-0096** [comment]
Hi @illsilin, a fix included in #612.
